app/lib/test_pals_mock_ldap.rb

```diff
@@ -0,0 +1,35 @@
+class TestPalsMockLDAP
+
+  TEST_PAL_NAMES = %w[
+    uri
+    rich
+    vivian
+    alonso
+    silva
+    laura
+    sarah
+    marcus
+    sofia
+    jodi
+    bill
+    hugo
+    fatima
+  ]
+
+  TEST_PAL_EMAILS = TEST_PAL_NAMES.map { |n| "#{n}@demo.studentinsights.org" }
+
+  def initialize(options)
+    @options = options
+    @email = options[:auth][:username]
+    @password = options[:auth][:password]
+  end
+
+  def bind
+    raise 'LDAP error' unless ::EnvironmentVariable.is_true('USE_TEST_PALS_LDAP')
+
+    return TEST_PAL_EMAILS.include?(@email) && @password == 'demo-password'
```

Comment:
Could this method read the database?   It seems like the list of usernames and passwords duplicates what's already there.  Or are you trying to simulate that the lists are actually different to test those cases?  I could see benefits either way.

Reply:
Oh interesting. I can see the benefit to reading the database for user emails. We won't be able to read passwords from the database anymore because we won't be storing passwords there.

Reply:
Yeah I was thinking that would still work because it's how the model worked before, but I agree removing them altogether would be better.  We should probably remove all those password fields from the model then too looks like there's quite a few.

And in https://github.com/studentinsights/studentinsights/pull/1701/files#r189039275 I was brainstorming other ways to store these passwords outside of code so that it would take a series of changes in order to disable normal production authentication (having multiple layers there seems important).